src/accounts/CancellationOverlay.tsx

```diff
@@ -0,0 +1,146 @@
+// Libraries
+import React, {FC, useContext, useMemo, useState} from 'react'
+import {
+  Overlay,
+  Alert,
+  ComponentColor,
+  ComponentSize,
+  IconFont,
+  Button,
+  ComponentStatus,
+} from '@influxdata/clockface'
+
+// Components
+import CancellationTerms from './CancellationTerms'
+import {
+  CancelServiceContext,
+  VariableItems,
+} from 'src/billing/components/PayAsYouGo/CancelServiceContext'
+import {track} from 'rudder-sdk-js'
+import {event} from 'src/cloud/utils/reporting'
+import {useSelector} from 'react-redux'
+import {getQuartzMe} from 'src/me/selectors'
+import {getOrg} from 'src/organizations/selectors'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+import {postSignout} from 'src/client'
+import {UserAccountContext} from './context/userAccount'
+
+interface Props {
+  onHideOverlay: () => void
+}
+
+const CancellationOverlay: FC<Props> = ({onHideOverlay}) => {
+  const {handleCancelAccount} = useContext(UserAccountContext)
+  const [hasAgreedToTerms, setHasAgreedToTerms] = useState(false)
+  const [hasClickedCancel, setHasClickedCancel] = useState(false)
+  const {
+    reason,
+    canContactForFeedback,
+    suggestions,
+    shortSuggestion,
+    getRedirectLocation,
+  } = useContext(CancelServiceContext)
+  const quartzMe = useSelector(getQuartzMe)
+  const org = useSelector(getOrg)
+
+  const handleCancelService = () => {
+    if (!hasClickedCancel) {
+      setHasClickedCancel(true)
+    }
+
+    const payload = {
+      org: org.id,
+      tier: quartzMe?.accountType,
+      email: quartzMe?.email,
+      alternativeProduct: shortSuggestion,
+      suggestions,
+      reason: VariableItems[reason],
+      canContactForFeedback: canContactForFeedback ? 'true' : 'false',
+    }
+    event('CancelServiceExecuted Event', payload)
+
+    if (
+      isFlagEnabled('rudderstackReporting') &&
+      isFlagEnabled('trackCancellations')
+    ) {
+      // Send to Rudderstack
+      track('CancelServiceExecuted', payload)
```

Comment:
is this copied over, or is this something we want to explicitly send to rudderstack? if you call `event` it should automatically send it to rudderstack

Reply:
This was copied over as well...